go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -0,0 +1,285 @@
+package ovn
+
+import (
+	"fmt"
+	"net"
+	"reflect"
+	"strings"
+	"sync"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	egressfirewallapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1"
+
+	"k8s.io/klog"
+)
+
+const (
+	defaultStartPriority = 2000
+)
+
+type egressFirewall struct {
+	sync.Mutex  //not sure if needed
+	name        string
+	namespace   string
+	egressRules []*egressFirewallRule
+}
+
+type egressFirewallRule struct {
+	id     int
+	access egressfirewallapi.EgressFirewallRuleType
+	ports  []*port
+	to     destination
+}
+
+type port struct {
+	protocol *string
+	portNum  *int32
+}
+
+type destination struct {
+	cidrSelector string
+}
+
+func newEgressFirewall(egressFirewallPolicy *egressfirewallapi.EgressFirewall) *egressFirewall {
```

Comment:
how about making this methods of EgressFirewall?

Reply:
I modelled this after network policy and made the functions similar and I made relevant functions methods of Controller. Should I make it different here?